### PR TITLE
Add /query endpoint

### PR DESCRIPTION
**Pull request by tjacovich**
Adds /query endpoint that allows adding and removing from libraries by passing a /search query.

POST
   - allows for adding/removing documents by passing /search params as POST data.
 
GET
   - Allows for adding documents by passing /search params in the url.

Updated unit tests to reflect new endpoint.

### CODE REVIEW

File: biblib/tests/unit_tests/test_views.py
Comment (kelockhart) [inline biblib/tests/unit_tests/test_views.py:1935]: Shouldn't the second part of this assert be `len(self.stub_library_2.bibcode)`, not the same but for `self.stub_library`?

File: biblib/tests/unit_tests/test_views.py
Comment (kelockhart) [inline biblib/tests/unit_tests/test_views.py:2020]: Same for this one - I'd rename `number_removed` here, since it's a dict

File: biblib/views/document_view.py
Comment (tjacovich) [reply biblib/views/document_view.py:648]: I think 2 makes sense as well. My thought is to merge the first PR and then pull those changes into here before making the final modifications.

**Review by kelockhart — COMMENTED** [review]
Looks good! We'll hold off on merging until you merge #159 and we can incorporate the merged `standard_ADS_bibcode_query` code, but no other changes needed here.

File: biblib/views/document_view.py
Comment (kelockhart) [inline biblib/views/document_view.py:753]: This docstring still needs to be updated to reflect that the post body is a query, not a list of bibcodes

File: biblib/views/document_view.py
Comment (kelockhart) [inline biblib/views/document_view.py:797]: Need to update to include the query (vs the bibcode list)

**Review by kelockhart — COMMENTED** [review]
Some cleanup to do here, but getting there. There were a few comments from last time that I'm not sure were addressed either - can you take a look at those?

File: biblib/tests/unit_tests/test_webservices.py
Comment (tjacovich) [reply biblib/tests/unit_tests/test_webservices.py:1222]: Added `test_query_cannot_add_library_without_q_get(self)`

File: biblib/tests/base.py
Comment (tjacovich) [reply biblib/tests/base.py:249]: Removed duplicate

File: biblib/views/document_view.py
Comment (tjacovich) [reply biblib/views/document_view.py]: Removed duplicate call to client

File: biblib/views/document_view.py
Comment (tjacovich) [reply biblib/views/document_view.py]: Added variables for readability

File: biblib/views/document_view.py
Comment (tjacovich) [reply biblib/views/document_view.py:752]: cleaned up doctring

File: biblib/tests/base.py
Comment (tjacovich) [reply biblib/tests/base.py:249]: Near as I can figure git decided they weren't the same and kept them both during a previous rebase. I can't find any difference between the two so I just picked one and deleted it.

File: biblib/views/document_view.py
Comment (kelockhart) [inline biblib/views/document_view.py:526]: We're logging the list of bibcodes returned twice - once here and once at line 514 above. For the one at line 514, I'd probably log something shorter, and leave this one, but up to you.

File: biblib/views/document_view.py
Comment (kelockhart) [inline biblib/views/document_view.py:533]: Ah and `valid_bibcodes` is getting logged here again. I'd pick one of these log statements to log the full list, and move the others to debug. Maybe leave a `len(valid_bibcodes` at the `info` level if needed

File: biblib/views/document_view.py
Comment (kelockhart) [inline biblib/views/document_view.py:576]: Same logging comment as above

**Review by kelockhart — APPROVED** [review]
Looks good! I had a few minor comments about logging, but feel free to merge once you've addressed those.

File: biblib/views/base_view.py
Comment (tjacovich) [reply biblib/views/base_view.py:29]: It's defined here because several functions use write access and most of them need this version of writes_allowed, but one version needs slightly different permissions. Everything else about the function is the same.

File: biblib/views/document_view.py
Comment (tjacovich) [reply biblib/views/document_view.py:526]: Opted to log just solr status code at 514

File: biblib/views/document_view.py
Comment (tjacovich) [reply biblib/views/document_view.py:533]: changed to debug logging.

File: biblib/views/base_view.py
Comment (tjacovich) [reply biblib/views/base_view.py:29]: `TransferView` needs different permissions so I defined them as a part of `BaseView` and the overwrite it in `TransferView`

**Pull request merged**
